### PR TITLE
Feat: Initial commit for AGW snap

### PR DESCRIPTION
**Pull request by Gmerold**

### CODE REVIEW

File: .github/workflows/agw.main.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: main-branch
+
+on:
+  pull_request:
```
Comment (gruyaume) [inline .github/workflows/agw.main.yml:4]: Can we also run it on pushes?

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+description = Run unit tests
+deps =
+    pytest
+    coverage[toml]
+    -r{toxinidir}/requirements.txt
+commands =
+    coverage run --source={[vars]src_path} -m pytest --ignore {[vars]integration_test_path} -v --tb native -s {posargs}
+    coverage report
+
+[testenv:integration]
```
Comment (gruyaume) [inline tox.ini:72]: This can be removed for now

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+# See LICENSE file for licensing details.
+[tox]
+skipsdist=True
+skip_missing_interpreters = True
+envlist = lint, static, unit
+
+[vars]
+src_path = {toxinidir}/magma_access_gateway_installer/
+unit_test_path = {toxinidir}/tests/unit/
+integration_test_path = {toxinidir}/tests/integration/
```
Comment (gruyaume) [inline tox.ini:11]: This can be removed for now

**Review by gruyaume — COMMENTED** [review]
Minor comments

File: .github/workflows/agw.main.yml (hunk 1 above)
Comment (Gmerold) [reply .github/workflows/agw.main.yml:4]: Yes, in fact I thought about the same in the evening.

File: tox.ini (hunk 2 above)
Comment (Gmerold) [reply tox.ini:72]: I removed it initially, but then I thought I can just as well create a placeholder for integration tests right away and have it ready for the future. 
If, for some reason, you think it's better to remove it for now - let me know.

File: tox.ini (hunk 3 above)
Comment (Gmerold) [reply tox.ini:11]: As above.

**Review by gruyaume — APPROVED** [review]

**Pull request merged**